src/Common/src/TypeSystem/IL/ILProvider.cs

```diff
@@ -175,6 +175,63 @@ private MethodIL TryGetPerInstantiationIntrinsicMethodIL(MethodDesc method)
                         }
                         break;
                     }
+                case "Array":
+                    {
+                        if (methodName == "GetComparerForReferenceTypesOnly")
```

Comment:
Nit: It would be nice to not have these intrinsics on Array. Array has a ton of random stuff in it. The less random stuff is dumped into it, the better. I think EqualityComparerHelpers would be a better place for them. (This would need to be adjusted in ProjectN too, of course.)